src/Compilers/Core/Portable/Collections/ImmutableArrayExtensions.cs

```diff
@@ -238,6 +238,21 @@ public static ImmutableArray<byte> ToImmutable(this MemoryStream stream)
             return builder.ToImmutableAndFree();
         }
 
+        /// <summary>
+        /// Maps an immutable array through a function that returns Tasks, returning the new ImmutableArray.
+        /// </summary>
+        public static async Task<ImmutableArray<TResult>> SelectAsArrayAsync<TItem, TResult>(this ImmutableArray<TItem> array, Func<TItem, Task<TResult>> selector)
```

Comment:
❔  The previous method (using `ValueTask`) is much better. Is there a reason we can't use it?

Reply:
I just create this one for because I hit the case when the nested func's return type is Task, if the ValueTask one is preferred I could edit the code later and remove this